File: src/testingil/webinar/tddlegacycode/e3/addfunctionality/s3/code/BaseCalculatorTests.java
package testingil.webinar.tddlegacycode.e3.addfunctionality.s3.code;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.BeforeEach;

public class BaseCalculatorTests {

	protected Calculator calc;
	protected String result;
	
	@BeforeEach
	public void setup(){
		calc = new Calculator();
	}
	
	protected BaseCalculatorTests pressing(String keys) {
		calc.press("C");
		keys.chars().forEach(c ->{
			calc.press(String.valueOf((char) c));
		}); 
		this.result = calc.getDisplay();
		return this;
	}

	protected BaseCalculatorTests atStart() {
		this.result = calc.getDisplay();
		return this;
	}

	protected void shouldShow(String expected) {
		assertEquals(expected, result);

	}

}

File: C#/E3.AddFunctionality/S1.Refactor/S1.guard/BaseCalculatorTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TDDLegacyCode.E3.AddFunctionality.S1.Refactor.S1.Guard
{
    public class BaseCalculatorTests
    {
        protected Calculator calc;
        protected string result;

        [TestInitialize]
        public void Setup()
        {
            calc = new Calculator();
        }

        protected BaseCalculatorTests Pressing(string keys)
        {
            calc.Press("C");
            foreach (char c in keys)
            {
                calc.Press(c.ToString());
            };
            this.result = calc.GetDisplay();
            return this;
        }

        protected BaseCalculatorTests AtStart()
        {
            this.result = calc.GetDisplay();
            return this;
        }

        public void ShouldShow(string expected)
        {
            Assert.AreEqual(expected, result);

        }

    }
}
